bin/lint_tool_file_names.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+# Prohibit internal shell script file names snake_case
+set -eo pipefail
+
+# find a list of files whose names do not match our convention
+errant_files=$(find bin/ _test/ -iname '*\.sh' -exec basename {} \; | grep '[^a-z_.].*')
+
+if [ -n "$errant_files" ]; then
+    echo "These file names do not follow our snake case convention:"
+    # find them again to print the whole relative path
+    while IFS= read -r file_name; do
+        find . -iname "$file_name"
```

Comment:
```suggestion
        find . -name "$file_name"
```

In this case we can be pretty confident that we have the named file's casing correct, and can avoid the case-conversion expense.